### PR TITLE
Centralize pagination/canonical meta URL generation in Document

### PR DESCRIPTION
**Pull request by askvortsov1**
Currently, our Document class does not support generating meta prev/next URLs. With this PR, content can set `page` and `hasNextPage` on the `Document` object. This, in turn, will appropriately generate next/prev links, and configure the canonical URL properly.

**Questions for Reviewers:**
As part of this, we introduce a helper function to add/override/remove params from URLs. Is this the best place for it?

As an alternative approach, we could create a generic `AbstractPaginatedContent` class with methods, and move logic into there. Then, Document would just have public fields for `nextPageUrl` and `prevPageUrl`. This would let Document focus on arrangement/presentation, and avoid hardcoding logic for a pagination scheme. On the other hand, if we have centralized, defined logic for a pagination scheme, we're less likely to run into issues.

### CODE REVIEW

**Comment by askvortsov1** [comment]
Before merging, we'll also want to append ": Page X" on non-first-pages to the page title.

File: src/Frontend/Document.php
Comment (askvortsov1) [reply src/Frontend/Document.php:299]: Agreed that this is out of place. Maybe using an external library, such as https://github.com/chroma-x/php-url-util would be good here? We don't really use plain function helpers in our PHP codebase at the moment.

Comment (SychO9) [reply src/Frontend/Document.php:299]: we could, though I don't know if we should easily require other packages though. Perhaps it'd be better then to leave it here and mark it protected.

**Review by SychO9 — APPROVED** [review]
Approving pending the title change mentioned in the comment above.

**Review by tankerkiller125 — APPROVED** [review]

**Comment by askvortsov1** [comment]
@tankerkiller125 @SychO9 re-requesting review, as I've refactored the basic title driver to use translations for layout.

Specific issues:
- Naming of translation keys
- The layout itself (did I put `: Page x` in the right place)

**Review by SychO9 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Pull request merged**
